mavros_extras/src/plugins/obstacle_distance.cpp

```diff
@@ -64,15 +64,32 @@ class ObstacleDistancePlugin : public plugin::PluginBase {
 	{
 		mavlink::common::msg::OBSTACLE_DISTANCE obstacle {};
 
-		auto n = std::min(req->ranges.size(), obstacle.distances.size());
-		auto cm_ranges = Eigen::Map<const Eigen::VectorXf>(req->ranges.data(), n) * 1e2;
-		Eigen::Map<Eigen::Matrix<uint16_t, Eigen::Dynamic, 1> > map_distances(obstacle.distances.data(), n);
+		if (req->ranges.size() <= obstacle.distances.size()) {
+			// all distances from sensor will fit in obstacle distance message
+			auto n = std::min(req->ranges.size(), obstacle.distances.size());
```

Comment:
Not needed anymore, req->ranges.size() will be always less than obstacle.arr.

Reply:
OK, removed "n" and substituted, "req->ranges.size()" wherever it was used.  To some extent I wanted to leave the existing code as it was to reduce worry but that's all good if you guys are happy.